src/server/session.ts

```diff
@@ -377,6 +377,67 @@ namespace ts.server {
             return perProjectResults;
         }
 
+        // Corner case: since FAR can cause projects to be loaded, it can cause files to be removed
+        // from inferred projects, possibly invalidating the results for those projects.
+
+        // NB: Iterate over a copy since we're going to modify the original as we go
+        for (const [project, referencedSymbols] of arrayFrom(perProjectResults.entries())) {
+            if (!project.dirty) {
+                continue;
+            }
+
+            const projectService = project.projectService;
+
+            let newReferencedSymbols: ReferencedSymbol[] | undefined;
+
+            for (let i = 0; i < referencedSymbols.length; i++) {
+                const referencedSymbol = referencedSymbols[i];
+                const def = referencedSymbol.definition;
+                const defScriptInfo = projectService.getScriptInfo(def.fileName)!;
+                if (!defScriptInfo.containingProjects.includes(project)) { // TODO: this check doesn't work correctly for dirty projects
+                    // If the def file is no longer in the project, the ref files must not be either -
+                    // otherwise, they'd pull the ref file in
+                    if (!newReferencedSymbols) {
+                        newReferencedSymbols = referencedSymbols.slice(0, i); // i.e. Don't include the current element
+                    }
+                    continue;
+                }
+
+                const references = referencedSymbol.references;
+                let newReferences: ReferencedSymbolEntry[] | undefined;
+                for (let j = 0; j < references.length; j++) {
+                    const ref = references[j];
+                    const refScriptInfo = projectService.getScriptInfo(ref.fileName)!;
+                    if (!refScriptInfo.containingProjects.includes(project)) { // TODO: this check doesn't work correctly for dirty projects

```

Comment:
@sheetalkamat I think I'm going to need some help with this.  I'm happy to explain offline what's going wrong, but I thought this might be a helpful way to show what I was attempting to do and give you a chance to debug the failure, if you're interested.  Can you help?

Reply:
https://github.com/microsoft/TypeScript/compare/refs?expand=1 isnt this correct way to handle this instead ?

Reply:
It's tempting, but I think that only covers the case where `isDefinition` needs to be updated in that project.

Another hacky solution might be to defer searching inferred projects until all (known?) configured projects have been searched.

Reply:
Yes but you also need to handle the references .. 

deferring inferred project isn’t ideal because you want to handle source maps and that could load more projects .. I think the right solution is to have guard against locations not part of project instead

Reply:
I think I've found a solution.